frontend/src/components/QuerySentencePreview/QuerySentencePreview.tsx

```diff
@@ -63,6 +63,7 @@ const QuerySentencePreview: FunctionComponent<QuerySentencePreviewProps> = (prop
   const [previewOption, setPreviewOption] = useState<PreviewOption>('config');
   const [alert, setAlert] = useState<string[]>([]);
   const [validOptions, setValidOptions] = useState<AdvancedQueryOptions>();
+  const [resetText, setResetText] = useState<string>('Clear');
```

Comment:
Give it a clear state (not `string`) for the expected options. Frankly, I'd rather you didn't use reset text as the state property for this as it seems particularly rigid. I suggest u change it to `resetStatus` with 2 options - `clear-all` or `clear-current`.
Enhance the button text to account for the active clause i.e `Clear Filter` or `Clear Join` e.t.c

Reply:
Updated variable naming and enhanced button text to account for active clause.